README.md

```diff
@@ -12,13 +12,66 @@ A user-friendly IRC bouncer.
 
 ## Usage
 
-    go run ./cmd/sojuctl create-user <username> -admin
+### Server side
+
+    go run ./cmd/sojuctl create-user <soju username> -admin
     go run ./cmd/soju -listen irc+insecure://127.0.0.1:6667
 
-Then connect with username `<username>/chat.freenode.net` and join `#soju`.
+### Client side
+
+soju can operate in two different modes: multi connection and single 
+connection.
+
+Even if you plan to use it in only one of these mode, you have to understand
+both to use soju properly. We will cover both modes in this tutorial.
+
+#### Single connection mode
+
+In this mode, 1 connection to a irc server = 1 connection to your soju bouncer.
+
+The easiest and fastest way to use this is to specify the address of the
+server in your username in your client configuration. For example to connect
+to freenode, your username will be: `<soju username>/chat.freenode.net`.
+Also set your soju password in the password field in your client configuration.
+
+This will autoconfigure soju by adding a network to the address 
+`chat.freenode.net` and the autoconnect to it. You will now be able to connect
+to any channel like you would normally do. soju will automatically save and 
+restore the channels you were connected to.
+
+#### Multi connection mode
+
+In this mode, a single connection to your soju bouncer can handle 
+multiple irc connections. You will need to manually configure
+each irc connection using the `BouncerServ`.
+
+Connect to your soju server by specifying your soju username in the
+username field in your client and your password in the password field.
+
+You should now be able to send private messages to the `BouncerServ`.
+You can send it commands to configure soju. Create new networks:
+
+```
+/msg BouncerServ network create -addr chat.freenode.net -name freenode
+/msg BouncerServ network create -addr irc.rizon.net:6697 -name rizon
+```
 
 See the man page at `doc/soju.1.scd` for more information.
 
+You will now be able to join channels on these networks by specifying their 
+name:
+
+```
+/join #soju/freenode
+/join #horriblesubs/rizon
+```
+
+#### Single connection mode using network name
```

Comment:
I would leave this out, because this piece of information isn't required for basic usage and it's a little confusing to have a "single upstream" section inside of the "multiple upstream" section. Users can connect with `<soju username>/chat.freenode.net` even if the network name is set to `freenode`.

Reply:
This is the most confusing part of soju imo so I think it might need more explanations than just what the man page provide. I'll make another section for this one.

Reply:
Networks can be referred to by their address or by their name, if any.

If it's confusing, users can just completely ignore it. The network name is purely optional.

Reply:
The thing that I find confusing is not the fact that there is two ways to refer to the network but more the fact that you need to use multi upstream mode to add extra configuration to your network to then connect using the single upstream mode. That why I made the tutorial step by step, I think it's a very incremental way to proceed in order to understand how to use it properly.

Reply:
Even if you're hitting the currently rare case where a network won't let you in unless you have a valid PASS, you can still use single-upstream mode.

1. Connect with username `<username>/<address>`
2. soju will fail to connect to the upstream server, and BouncerServ will tell you the error. 
3. Send a BouncerServ command to set the PASS: `/msg BouncerServ net update <address> -pass <pass>`
4. soju will try to reconnect to the upstream server with the PASS